src/decorators/registry.ts

```diff
@@ -0,0 +1,25 @@
+import { handleDecorator } from './handleDecorator';
+import { RegistryItem } from '../Registry';
+
+export interface RegistryConfig {
+	[ name: string ]: RegistryItem;
+}
+
+/**
+ * Decorator that can be used to register a widget with the registry
+ */
+export function registry(nameOrConfig: string | RegistryConfig, loader?: RegistryItem) {
+	return handleDecorator((target, propertyKey) => {
+		target.addDecorator('constructor', function(this: any) {
+			if (typeof nameOrConfig === 'string') {
+				this.registry.define(nameOrConfig, loader);
+			} else {
+				Object.keys(nameOrConfig).forEach(name => {
```

Comment:
Until we move to prettier, at the moment the style guide is for `name` to be surrounded by parentheses `(name)`

Reply:
Not a problem, I thought we'd already dropped brackets for single params.